lib/utils/index.test.ts

```diff
@@ -0,0 +1,57 @@
+import { isFunction, isNumber, isString, isUndefined } from "./index";
+
+describe("isUndefined", () => {
+  const tests = [
+    { input: 42, expected: false },
+    { input: "a string", expected: false },
+    { input: undefined, expected: true },
+    { input: () => null, expected: false }
+  ];
+  for (const { input, expected } of tests) {
+    it(`should return ${expected} when passed ${input}`, () => {
```

Comment:
You can use a Jest [`each` ](https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout).